www/js/survey/enketo/UserInputButton.tsx

```diff
@@ -15,18 +15,34 @@ import { useTheme } from 'react-native-paper';
 import { displayErrorMsg, logDebug } from '../../plugin/logger';
 import EnketoModal from './EnketoModal';
 import LabelTabContext from '../../diary/LabelTabContext';
+import useAppConfig from '../../useAppConfig';
+import { getSurveyForTimelineEntry } from './conditionalSurveys';
 
 type Props = {
   timelineEntry: any;
 };
 const UserInputButton = ({ timelineEntry }: Props) => {
   const { colors } = useTheme();
+  const appConfig = useAppConfig();
   const { t, i18n } = useTranslation();
 
   const [prevSurveyResponse, setPrevSurveyResponse] = useState<string | undefined>(undefined);
   const [modalVisible, setModalVisible] = useState(false);
   const { userInputFor, addUserInputToEntry } = useContext(LabelTabContext);
 
+  // which survey will this button launch?
+  const [surveyName, notFilledInLabel] = useMemo(() => {
+    const tripLabelConfig = appConfig?.survey_info?.buttons?.['trip-label'];
+    if (!tripLabelConfig) {
+      // config doesn't specify; use default
+      return ['TripConfirmSurvey', t('diary.choose-survey')];
+    }
```

Comment:
Would it ever make sense to never have a trip-level survey? Similar to the argument for the `null` match, maybe we want to only show place surveys and not trip surveys? I see that this is what we do now, which is why the time use displays the `TripConfirmSurvey` although there is no button configured for it, but we might want to revisit that later